controlled_navigation/controlled_navigation.py

```diff
@@ -59,31 +117,170 @@ def author_view(self, context):
+        """
+        if self.randomness:
+            if not self.randomized_children_ids:
+                self.generate_randomized_children_ids()
+            return self.randomized_children_ids
+        return self.get_parsed_children_ids()
+
+    def generate_randomized_children_ids(self) -> None:
+        """
+        Generate a randomized list of children ids.
+        """
+        self.randomized_children_ids = [str(child) for child in self.children]
+        random.shuffle(self.randomized_children_ids)
+
+    def get_parsed_children_ids(self) -> List[str]:
+        """
+        Get the parsed children of this block.
+
+        `self.children` is a list of the children's UsageKeys. This method returns
+        a list of the children's IDs, converting each UsageKey to a string.
+
+        Returns:
+            List[str]: List of children ids.
+        """
+        return [str(child) for child in self.children]
+
+    def is_first_child(self, children_ids: List[str]) -> bool:
+        """
+        Check if the current child is the first one.
+
+        Args:
+            children_ids (List[str]): List of children ids.
+
+        Returns:
+            bool: If the current child is the first one.
+        """
+        return self.current_child_id == str(children_ids[0])
+
+    def is_last_child(self, children_ids: List[str]) -> bool:
+        """
+        Check if the current child is the last one.
+
+        Args:
+            children_ids (List[str]): List of children ids.
+
+        Returns:
+            bool: If the current child is the last one.
+        """
+        return self.current_child_id == str(children_ids[-1])
+
+    @XBlock.json_handler
+    def next_child(
+        self, data: dict, suffix: str = ""  # pylint: disable=unused-argument
+    ) -> dict:
+        """
+        Render the next child of the XBlock.
+
+        Args:
+            data (dict): Optional data.
+            suffix (str, optional): Suffix. Defaults to "".
+
+        Returns:
+            dict: If the request was successful.
+        """
+        children = self.get_children_ids()
+        current_index = children.index(self.get_current_child_id())
+        if current_index == len(children) - 1:
+            return {"result": "error"}
+        self.current_child_id = children[current_index + 1]
+        return {"result": "success"}
+
+    @XBlock.json_handler
+    def prev_child(
+        self, data: dict, suffix: str = ""  # pylint: disable=unused-argument
+    ) -> dict:
+        """
+        Render the previous child of the XBlock.
+
+        Args:
+            data (dict): Optional data.
+            suffix (str, optional): Suffix. Defaults to "".
+
+        Returns:
+            dict: If the request was successful.
+        """
+        children = self.get_children_ids()
+        current_index = children.index(self.get_current_child_id())
+        if current_index == 0:
+            return {"result": "error"}
```

Comment:
[curious] Why is this handled as an error?

Reply:
It indicates that you cannot go back because there are no more children. The same for when you want to move forward.